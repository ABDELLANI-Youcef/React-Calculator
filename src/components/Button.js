import PropTypes from 'prop-types';

const Button = ({ name, clickHandler }) => {
  const handleClick = () => {
    clickHandler(name);
  };
  return (
    <button type="button" onClick={handleClick}>{name}</button>
  );
};
Button.propTypes = {
  name: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default Button;
