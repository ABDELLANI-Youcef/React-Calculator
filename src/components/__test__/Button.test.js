import { render, screen } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

test('renders learn react link', () => {
  render(
    <Button name="+" wide={false} color clickHandler={() => true} />,
  );
  const linkElement = screen.getByText(/\+/i);
  expect(linkElement).toBeInTheDocument();
});

it('should render button correct', () => {
  const button = renderer.create(
    <Button name="+" wide={false} color clickHandler={() => true} />,
  ).toJSON();
  expect(button).toMatchSnapshot();
});

it('shoud make the width 50% if the wide attribut is true', () => {
  const button = renderer.create(
    <Button name="0" wide color={false} clickHandler={() => true} />,
  ).toJSON();
  expect(button).toMatchSnapshot();
});

it('should make the color of button white if the color is false', ()=>{
  const button = renderer.create(
    <Button name="AC" wide={false} color={false} clickHandler={() => true} />,
  ).toJSON();
  expect(button).toMatchSnapshot();
});