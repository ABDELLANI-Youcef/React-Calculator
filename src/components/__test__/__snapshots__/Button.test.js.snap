// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoud make the width 50% if the wide attribut is true 1`] = `
<button
  className="button"
  data-testid="0"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "width": "50%",
    }
  }
  type="button"
>
  0
</button>
`;

exports[`should make the color of button white if the color is false 1`] = `
<button
  className="button"
  data-testid="AC"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "width": "25%",
    }
  }
  type="button"
>
  AC
</button>
`;

exports[`should render button correct 1`] = `
<button
  className="button"
  data-testid="+"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "orange",
      "width": "25%",
    }
  }
  type="button"
>
  +
</button>
`;
